rocAL/source/augmentations/color_augmentations/node_color_cast.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <vx_ext_rpp.h>
+#include "augmentations/color_augmentations/node_color_cast.h"
+#include "pipeline/exception.h"
+
+namespace {
+static void fill_rgb_for_batch(std::vector<float> &rgb_out, unsigned batch_size, const std::vector<float> &rgb_in) {
+    rgb_out.resize(batch_size * 3);
+    if (rgb_in.size() == 3) {
+        // Replicate a single triplet across the batch
+        for (unsigned i = 0; i < batch_size; ++i) {
+            unsigned base = i * 3;
+            rgb_out[base + 0] = rgb_in[0];
+            rgb_out[base + 1] = rgb_in[1];
+            rgb_out[base + 2] = rgb_in[2];
+        }
+    } else if (rgb_in.size() == batch_size * 3) {
+        // Copy per-sample triplets
+        rgb_out = rgb_in;
+    } else {
+        // Invalid size, default to zeros
```

Comment:
Blocking: When rgb_in.size() does not match 3 or batch_size*3 we silently zero out the color cast weights. That turns the augmentation into a no-op without any signal to the user, which makes configuration bugs extremely hard to track down. Could we fail fast here (e.g. throw or at least log an error) instead of muting the error?